src/wp-includes/block-template-utils.php

```diff
@@ -1432,39 +1432,45 @@ function get_template_hierarchy( $slug, $is_custom = false, $template_prefix = '
 	$template_hierarchy[] = 'index';
 	return $template_hierarchy;
 }
+
 /**
  * Inject ignoredHookedBlocks metadata attributes into a template or template part.
  *
- * Given a `wp_template` or `wp_template_part` post object, locate all blocks that have
+ * Given an object that represents a `wp_template` or `wp_template_part` post object
+ * prepared for inserting or updating the database, locate all blocks that have
  * hooked blocks, and inject a `metadata.ignoredHookedBlocks` attribute into the anchor
  * blocks to reflect the latter.
  *
- * @param WP_Post $post A post object with post type set to `wp_template` or `wp_template_part`.
- * @return WP_Post The updated post object.
+ * @param stdClass        $post    An object representing a template or template part
+ *                                 prepared for inserting or updating the database.
+ * @param WP_REST_Request $request Request object.
+ * @return stdClass The updated object representing a template or template part.
  */
-function inject_ignored_hooked_blocks_metadata_attributes( $post ) {
+function inject_ignored_hooked_blocks_metadata_attributes( $post, $request ) {
```

Comment:
While we're at it: please add `@since` annotation to this function too.

cc @ockham

Reply:
Probably want to add `@access private` too because extenders shouldn't really be using this function

Reply:
Done in https://github.com/WordPress/wordpress-develop/pull/6225/commits/9e84d934c44f6bbc585a9d230c7f629302750778 and https://github.com/WordPress/wordpress-develop/pull/6225/commits/ad3d547cb60798dc769326527f59500490cd65e1